File: AD-HOC/1031/main.dart
import 'dart:io';

void main(List<String> args) {
  int n = int.parse(stdin.readLineSync()!);
  while (n != 0) {
    for (int m = 1; m > 0; m++) {
      List<int> regions = [for (int i = 1; i <= n; i++) i];
      int selected = 0;

      while (regions.length > 1) {
        int region = regions.removeAt(selected);
        if (region == 13) break;
        selected = (selected + m - 1) % regions.length;
      }

      if (regions.last == 13) {
        print(m);
        break;
      }
    }
    n = int.parse(stdin.readLineSync()!);
  }
}

File: AD-HOC/1031/main.rb
# frozen_string_literal: true

while (n = gets.to_i) != 0
  (1..).each do |m|
    regions = [*1..n]
    selected = 0
    while regions.length > 1
      region = regions.delete_at(selected)
      break if region == 13

      selected = (selected + m - 1) % regions.length
    end
    if regions.last == 13
      puts m
      break
    end
  end
end
